File: app/src/main/java/com/trimettransit/tracker/data/model/Detour.java
package com.trimettransit.tracker.data.model;

import android.os.Parcel;
import android.os.Parcelable;

public class Detour implements Parcelable {
    public static final Parcelable.Creator CREATOR = new Creator();

    private int id;
    private String desc;
    private int[] routes;

    static class Creator implements Parcelable.Creator {
        Creator() {
        }

        @Override // android.os.Parcelable.Creator
        public Object createFromParcel(Parcel parcel) {
            return new Detour(parcel, null);
        }

        @Override // android.os.Parcelable.Creator
        public Object[] newArray(int i) {
            return new Detour[i];
        }
    }

    /* synthetic */ Detour(Parcel parcel, Creator creator) {
        this(parcel);
    }

    public int getId() {
        return this.id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getDesc() {
        return this.desc;
    }

    @Override // android.os.Parcelable
    public int describeContents() {
        return 0;
    }

    @Override // android.os.Parcelable
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeInt(this.id);
        parcel.writeString(this.desc);
        parcel.writeIntArray(this.routes);
    }

    public Detour() {
    }

    public void setDesc(String desc) {
        this.desc = desc;
    }

    private Detour(Parcel parcel) {
        this.id = parcel.readInt();
        this.desc = parcel.readString();
        this.routes = parcel.createIntArray();
    }

    public int[] getRoutes() {
        return this.routes;
    }

    public void setRoutes(int[] routes) {
        this.routes = routes;
    }
}

File: common/model/src/main/java/com/trimettransit/tracker/model/Detour.kt
package com.trimettransit.tracker.model

import android.os.Parcel
import android.os.Parcelable

data class Detour(
    val id: Int = 0,
    val desc: String = "",
    val routes: List<Int>? = null
) : Parcelable {

    constructor(parcel: Parcel) : this(
        parcel.readInt(),
        parcel.readString() ?: "",
        parcel.createIntArray()?.toList()
    )

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(id)
        parcel.writeString(desc)
        parcel.writeIntArray(routes?.toIntArray() ?: intArrayOf())
    }

    override fun describeContents(): Int = 0

    companion object {
        @JvmField
        val CREATOR = object : Parcelable.Creator<Detour> {
            override fun createFromParcel(parcel: Parcel): Detour = Detour(parcel)
            override fun newArray(size: Int): Array<Detour?> = arrayOfNulls(size)
        }
    }
}
